src/pagination/__tests__/stateful-container.test.js

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+/* eslint-env browser */
+import React from 'react';
+import {mount} from 'enzyme';
+import {STATE_CHANGE_TYPE} from '../constants';
+import StatefulContainer from '../stateful-container';
+
+const mockChildrenFn = jest.fn().mockImplementation(() => <div />);
+const mockOnPageChangeFn = jest.fn();
+const mockStateReducerFn = jest
+  .fn()
+  .mockImplementation((changeType, changes) => changes);
+
+function getSharedProps() {
+  return {
+    numPages: 7,
+    children: mockChildrenFn,
+    onPageChange: mockOnPageChangeFn,
+    stateReducer: mockStateReducerFn,
+  };
+}
+
+describe('Pagination StatefulContainer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('renders and passes required props to children function', () => {
+    const component = mount(<StatefulContainer {...getSharedProps()} />);
+    expect(mockChildrenFn.mock.calls[0][0]).toEqual({
+      currentPage: 1,
+      onPageChange: component.instance().onPageChange,
+    });
+  });
+
+  test('with initialState', () => {
+    const props = {
+      ...getSharedProps(),
+      initialState: {
+        currentPage: 5,
+      },
+    };
+    const component = mount(<StatefulContainer {...props} />);
+    expect(component.state('currentPage')).toBe(5);
+  });
+
+  test('onPageChange', () => {
```

Comment:
there are different human friendly names for unit test with explanation, why it's needed.
This one doesn't seems to comply with any of those. It contains 3 `expect`, which do not belong to one logical `assert`, they check for different things
https://dzone.com/articles/7-popular-unit-test-naming

Reply:
thanks for calling out my laziness :p, will update the name to be clearer here